crates/ra_hir_ty/src/unsafe_validation.rs

```diff
@@ -0,0 +1,138 @@
+//! Provides validations for unsafe code. Currently checks if unsafe functions are missing
+//! unsafe blocks.
+
+use std::sync::Arc;
+
+use hir_def::{DefWithBodyId, FunctionId};
+use hir_expand::diagnostics::DiagnosticSink;
+
+use crate::{
+    db::HirDatabase, diagnostics::MissingUnsafe, lower::CallableDef, ApplicationTy,
+    InferenceResult, Ty, TypeCtor,
+};
+
+use rustc_hash::FxHashSet;
+
+use hir_def::{
```

Comment:
https://github.com/rust-analyzer/rust-analyzer/blob/master/docs/dev/README.md#order-of-imports

Reply:
Done